app/controllers/tradingpremises/AgentNameController.scala

```diff
@@ -126,12 +126,17 @@ class AgentNameController @Inject()(
                 _ <- updateDataStrict[TradingPremises](request.credId, index) { tp =>
                   tp.agentName(tradingPremises.agentName.get.copy(dateOfChange = Some(dateOfChange)))
                 }
-              } yield Redirect(routes.DetailedAnswersController.get(1))
+              } yield Redirect(routes.DetailedAnswersController.get(index))
           }
         }
   }
 
   private def redirectToAgentNameDateOfChange(tradingPremises: TradingPremises, name: AgentName) = {
-    !tradingPremises.agentName.contains(name) && tradingPremises.lineId.isDefined
+    def isAgentNameAndDobChanged = tradingPremises.agentName match {
+      case Some(AgentName(agentName, _, agentDateOfBirth)) if agentName == name.agentName && agentDateOfBirth == name.agentDateOfBirth => false
```

Comment:
Consider putting the following in a function;

`agentName == name.agentName && agentDateOfBirth == name.agentDateOfBirth`

Reply:
sure done :)